File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        intervals.sort(key=lambda x:x[0])

        merge = [intervals[0]]
        
        for current in intervals[1:]:

            last = merge[-1]

            if current[0]<= last[-1]:
                merge[-1] = [last[0],max(last[-1],current[-1])]
            else:
                merge.append(current)
        return merge

File: 56-merge-intervals/merge-intervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        List<int[]> merged = new ArrayList<>();
        merged.add(intervals[0]);
        
        for (int i = 1; i < intervals.length; i++) {
            int[] current = intervals[i];
            int[] last = merged.get(merged.size() - 1);
            
            if (current[0] <= last[1]) {
                merged.set(merged.size() - 1, new int[]{last[0], Math.max(last[1], current[1])});
            } else {
                merged.add(current);
            }
        }
        
        return merged.toArray(new int[merged.size()][]);
    }
}

